pkg/kapp/clusterapply/add_or_update_change.go

```diff
@@ -25,6 +26,10 @@ const (
 	updateStrategySkipAnnValue              ClusterChangeApplyStrategyOp = "skip"
 )
 
+var (
+	resourceWithHistoryDebug = os.Getenv("KAPP_DEBUG_RESOURCE_WITH_HISTORY") == "true"
```

Comment:
`resourceWithHistoryDebug` is used to debug the hashing and ops diffing we do while deciding what kind of diff we do.
Is there a particular reason why we are printing the warning only in the debug mode and not for our users?

Reply:
I think this is because the users didn't much care about the warnings shown. But this is debatable if others also think that it is good to show a warning by default then I can make that change. 
(Keeping this unresolved to get the inputs)

Reply:
I definitely think that we should let the users know, as this behaviour actually results in them seeing diffs they otherwise would not have.